lib/strings.nix

```diff
@@ -144,6 +144,19 @@ rec {
   */
   concatLines = concatMapStrings (s: s + "\n");
 
+  /* Replicate a string n times, and concatenate the parts into
+     a new string.
+
+  Type: replicateString :: int -> string -> string
+
+    Example:
+      replicateString 3 "v"
+      => "vvv"
+      replicateString 5 "hello"
+      => "hellohellohellohellohello"
+  */
+  replicateString = n: s: concatStrings (lib.replicate n s);
```

Comment:
Hmm how about just calling this `replicate` and not exposing it under `lib.*`? Then users can still use `lib.strings.replicate` instead of `lib.replicateString`.

Reply:
Generally `lib` is moving towards not exposing all functions in the top level.

Reply:
I'd be okay with that. I see you commented on this in #263969 as well. Do you by any chance have an issue or discourse post, so I could keep track on this?

Reply:
Opened https://github.com/NixOS/nixpkgs/issues/266103 :)